src/data/W2lListFilesDataset.cpp

```diff
@@ -78,7 +78,12 @@ std::vector<W2lLoaderData> W2lListFilesDataset::getLoaderData(
     }
 
     data[id].sampleId = data_[i].getSampleId();
-    data[id].input = loadSound(data_[i].getAudioFile());
+    auto path = data_[i].getAudioFile();
+    try {
+      data[id].input = loadSound(path);
+    } catch (...) {
```

Comment:
Since the exception messages in `loadSound` are informative, let's catch `const std::exception& ex` first and pass `ex.what()` to the new exception message before catching all